modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java

```diff
@@ -338,93 +338,98 @@ private void updateSlotAssignmentMap(String queueName, Slot allocatedSlot, Strin
+                                log.debug("RightExtra in overlapping slot : " + rightExtraSlot);
+                            }
+
+                            //Update last message ID - expand ongoing slot to cater this leftover part.
+                            messageIdSet.add(lastMessageIdInTheSlot);
+                            wrapper.setLongTreeSet(messageIdSet);
+                            slotIDMap.set(queueName, wrapper);
+                            if (log.isDebugEnabled()) {
+                                log.debug(lastMessageIdInTheSlot + " added to slotIdMap " +
+                                       "(RightExtraSlot). Current values in " +
+                                    "map " + messageIdSet);
+                            }
+
+                            //record last published message id
+                            nodeToLastPublishedIDMap.set(nodeId, lastMessageIdInTheSlot);
                         }
 
-                        //record last published message id
-                        nodeToLastPublishedIDMap.set(nodeId, lastMessageIdInTheSlot);
-                    }
 
 
+                    }
 
-                }
+                } else {
+                    /**
+                     * Update the slotIDMap only if the last assigned message ID is less than the new start message ID
+                     */
+                    messageIdSet.add(lastMessageIdInTheSlot);
+                    wrapper.setLongTreeSet(messageIdSet);
+                    slotIDMap.set(queueName, wrapper);
+                    if (log.isDebugEnabled()) {
+                        log.debug("No overlapping slots found during slot submit "+ startMessageIdInTheSlot + " to : " +
+                                lastMessageIdInTheSlot+". Added msgID " +
+                                lastMessageIdInTheSlot + " to slotIDMap");
+                    }
 
-            } else {
-                /**
-                 * Update the slotIDMap only if the last assigned message ID is less than the new start message ID
-                 */
-                messageIdSet.add(lastMessageIdInTheSlot);
-                wrapper.setLongTreeSet(messageIdSet);
-                slotIDMap.set(queueName, wrapper);
-                if (log.isDebugEnabled()) {
-                    log.debug("No overlapping slots found during slot submit "+ startMessageIdInTheSlot + " to : " +
-                            lastMessageIdInTheSlot+". Added msgID " +
-                            lastMessageIdInTheSlot + " to slotIDMap");
+                    //record last published message ID
+                    nodeToLastPublishedIDMap.set(nodeId, lastMessageIdInTheSlot);
                 }
-
-                //record last published message ID
-                nodeToLastPublishedIDMap.set(nodeId, lastMessageIdInTheSlot);
             }
+        } else {
+            log.warn("The server is shutting down and therefore the cluster is inactive. Therefore updating the slot " +
```

Comment:
Fixed as suggested. The error will only be printed if the server is not in shutting_down state. If it is, the log is only printed in debug mode.